dear_bindings.py

```diff
@@ -218,7 +219,10 @@ def convert_header(
     mod_flatten_namespaces.apply(dom_root, {'ImGui': 'ImGui_'})
     mod_flatten_nested_classes.apply(dom_root)
     # The custom type fudge here is a workaround for how template parameters are expanded
-    mod_flatten_templates.apply(dom_root, custom_type_fudges={'const ImFont**': 'ImFont* const*'})
+    # Each iteration handles templates one more nesting level deep
+    for _ in range(0, 2):
+        mod_flatten_templates.apply(dom_root, custom_type_fudges={'const ImFont**': 'ImFont* const*'})
```

Comment:
All of the changes I've done were a result of eliminating the parsing/generation/compilation errors one by one, and this is no exception. It is used, although exactly once, to handle this type definition:
```
template<typename T>
struct ImPool
{
    ImVector<T>     Buf;        // Contiguous data
...
}
```

I agree this isn't elegant in the slightest, but since this is only needed to handle that one case, I figured it's probably fine to get us to a compileable state.

Reply:
Oh, whoops - sorry, I thought I'd checked for anything that was actually getting touched by that but I must have missed that case!

In that case, yeah, you're right and we need this code.
So I think the two big things that strike me are:

1) It would probably be better to put the loop inside the filter, keep track of if anything was actually found that needed flattening, and then loop until nothing more was found to do. This isn't really time-critical code so I figure we can afford the extra pass over the data if it makes things more robust and easier to read.

2) The direct references to tokens by index (`unmodified_element.tokens[0]` and `template_parameter_tokens[1].value` make me nervous - what guarantees do we have that the tokens we expect appear in those locations (for example, would adding/removing a `typename` somewhere cause them to shift?). If there's a way the code can be changed to do at least a basic scan to locate the tokens it's interested in then that would probably make it a lot more robust (I realise that the original code isn't exactly hugely well-behaved in that regard, but it does at least check for a few of those cases).

If that's fiddly then some comments around what those magic numbers represent would probably go a long way to making it easier to debug if something does go wrong. (In general any extra comments that could be added in that function would probably be a big win - it's pretty complicated and the more explanation there is the better, I suspect)

Reply:
Did the changes as you suggested. Still a bit wobbly, but much better than it was!